cmd/restic/lock.go

```diff
@@ -32,7 +32,39 @@ func lockRepository(ctx context.Context, repo *repository.Repository, exclusive
 		lockFn = restic.NewExclusiveLock
 	}
 
-	lock, err := lockFn(ctx, repo)
+	retrySleep := 10 * time.Second
+	if retrySleep > globalOptions.WaitLock/4 {
+		// Reduce the time between the retries if the maximum total wait
+		// duration is small so that there are at least a few attempts.
+		retrySleep = globalOptions.WaitLock / 4
+	}
+
+	var lock *restic.Lock
+	var err error
+	waitMessagePrinted := false
+	timeout := time.After(globalOptions.WaitLock)
+retryLoop:
+	for {
+		lock, err = lockFn(ctx, repo)
+		if err != nil && restic.IsAlreadyLocked(err) {
+			// repo already locked
+			select {
+			case <-timeout:
```

Comment:
I'd prefer a slightly different retry logic:
Try to lock the repository for the first time, then sleep for `retrySleep` or until `timeout` has expired whichever happens earlier, then try to lock the repository again. If timeout has expired this is the last try, otherwise loop and sleep again. That would avoid when waiting for e.g. 55 seconds that the last 5 seconds have no use at all.